v2/sourcedb-to-spanner/src/test/java/com/google/cloud/teleport/v2/utilities/ShardedMySQLResourceManager.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (C) 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.teleport.v2.utilities;
+
+import org.apache.beam.it.jdbc.AbstractJDBCResourceManager;
+import org.apache.beam.vendor.guava.v32_1_2_jre.com.google.common.annotations.VisibleForTesting;
+import org.testcontainers.containers.MySQLContainer;
+import org.testcontainers.utility.DockerImageName;
+
+/**
+ * Mysql resource manager to be used for sharded migrations. This class has been created as the
+ * MySQL resource manager constructor is not public.
```

Comment:
Why do we need to make it public? Do we plan on running multiple logical shards on different ports? 
If yes, is it not easier to build a wrapper that can spawn a list of MySQLResourceManagers.
Or simply just instantiate multiple MySQLResourceManagers as fields in the IT class itself, we will hardly require more than 2-3 for ITs)

Reply:
The resource managers all run on port 3306, it causes a clash.

Reply:
This has been removed.